### DISCUSSION TITLE
output-location with html

### DISCUSSION THREAD

**Discussion by Walser52** [quarto-dev/quarto-cli#10758] [Q&A]
### Description

Quarto's documentation for presentation shows there is an `output-location` option for revealjs. Is there a similar option for html output? I have a website with python code and I would like plots to be rendered side by side. 

I can try to store the result in a variable and then use a two column setup with the code in column A, and `{python} result` in column B.

Is there a cleaner way?

**Comment by cderv** [comment]
`output-location` is only available for revealjs format today. This could be an interesting feature request to make this work for other outputs. 🤔 

I guess it would be leveraging block layout: https://quarto.org/docs/authoring/figures.html#block-layout

With latest Quarto 1.6 pre-release, we added a feature to reuse code cell output. So you could now to that manually. See pre-release doc: https://prerelease.quarto.org/docs/authoring/contents.html

Something like 
````markdown
---
title: Layout plot / code
format: html
---

## A section

:::: {layout="[ 40, 60 ]"}

::: {#first-column}

Here we define a plot.

```{python}
#| label: a-cell
import matplotlib.pyplot as plt
plt.plot([1,2,3])
```

:::

::: {#second-column}

See the plot

{{< contents a-cell >}}

:::

::::
````

However, it seems we have an issue with Python cell and layout - we put the code cell outside always ! 

![image](https://github.com/user-attachments/assets/7626686f-c973-4483-b158-6f3fb216b90e)

````markdown
---
title: Layout plot / code
format: html
---

## A section

:::: {layout="[ 40, 60 ]"}

::: {#first-column}

Here we define a plot.

```{python}
#| label: a-cell
import matplotlib.pyplot as plt
plt.plot([1,2,3])
```

:::

::: {#second-column}

Comment

:::

::::

````

![image](https://github.com/user-attachments/assets/29f59a2c-dd59-4f13-a25e-a7b692c9ee37)

This is surprising to me - but I remember a discussion about this somewhere. I'll find it.

@cscheid would you have expected this to be a good usage of `contents` shortcode ?

**Reply by mcanouil** [reply]
FYI, there is a Quarto extension for something similar but as panel: https://github.com/coatless-quarto/panelize

- https://github.com/quarto-dev/quarto-cli/discussions/9646#discussioncomment-9451134

**Reply by cderv** [reply]
> This is surprising to me - but I remember a discussion about this somewhere. I'll find it.

And here it is : 
- https://github.com/quarto-dev/quarto-cli/issues/8179

I think I am as surprised by this behavior now as I was back then. It feels really not expected to me that code cells is move out of a column when the Markdown syntax really express it to have the code cell in first column. Anyhow, not the place to discuss. 


Regarding my proposal, it does not work as I think it was. Even using `.columns` does not help because code cell and output are moved together.
````markdown
---
title: Layout plot / code
format: html
---

## A section

:::: {.columns}

::: {.column width=40%}

Here we define a plot.

```{python}
#| label: a-cell
import matplotlib.pyplot as plt
plt.plot([1,2,3])
```

:::

::: {.column width=60%}

See the plot

{{< contents a-cell >}}

:::

::::

````

![image](https://github.com/user-attachments/assets/7aee2e9d-8d4a-415f-aab8-be443992d23b)

To make it works with column, we would need the output only to be inserted inside a div with a label we can refer. 

I'll post a working work around in another post below

**Reply by mcanouil** [reply]
I edited to add the comment from Carlos that initiated the extension (Lua filter)
- https://github.com/quarto-dev/quarto-cli/discussions/9646#discussioncomment-9451134

**Comment by cderv** [comment]
possible workaround - just sharing for reference 


Duplicating the code cell and have one evaluate while the other no would make this possible. Not ideal. 
Maybe an extension like panelize would make this easier to use.

````markdown
---
title: Layout plot / code
format: html
---

## A section

:::: {.columns}

::: {.column width=40%}

Here we define a plot.

```{python}
#| eval: false
import matplotlib.pyplot as plt
plt.plot([1,2,3])
```

:::

::: {.column width=60%}

See the plot

```{python}
#| echo: false
#| label: a-cell
import matplotlib.pyplot as plt
plt.plot([1,2,3])
```


:::

::::

````

![image](https://github.com/user-attachments/assets/fe6e11f9-8d16-4fb4-9e97-db2f0973bd43)

**Reply by cderv** [reply]
For completeness, we are limited by Jupyter engine here. **knitr** does have mecanism to reuse code cell easily 

````markdown
---
title: Layout plot / code
format: html
---

## A section

:::: {.columns}

::: {.column width=40%}

Here we define a plot.

```{r}
#| eval: false
#| label: a-cell
plot(cars)
```

:::

::: {.column width=60%}

See the plot

```{r}
#| echo: false
#| label: a-cell
```


:::

::::
````
